pilot/pkg/security/authz/model/util.go

```diff
@@ -109,3 +109,37 @@ func found(key string, list []string) bool {
 	}
 	return false
 }
+
+// getIdentityWithNewTrustDomain returns a new SPIFFE identity with the new trust domain.
+// The trust domain corresponds to the trust root of a system.
+// Refer to
+// [SPIFFE-ID](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain)
+// In Istio authorization, an identity is presented in the format:
+// <trust-domain>/ns/<some-namespace>/sa/<some-service-account>
+func getIdentityWithNewTrustDomain(trustDomain, spiffeIdentity string) string {
+	identityParts := strings.Split(spiffeIdentity, "/")
+	// A valid SPIFFE identity in authorization has no SPIFFE:// prefix.
+	// It is presented as <trust-domain>/ns/<some-namespace>/sa/<some-service-account>
+	if len(identityParts) != 5 {
+		rbacLog.Errorf("Wrong SPIFFE format found: %s", spiffeIdentity)
+		return ""
+	}
+	return fmt.Sprintf("%s/%s", trustDomain, strings.Join(identityParts[1:], "/"))
+}
+
+// The trust domain corresponds to the trust root of a system.
+// Refer to
+// [SPIFFE-ID](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain)
+// In Istio authorization, an identity is presented in the format:
+// <trust-domain>/ns/<some-namespace>/sa/<some-service-account>
+// Therefore, for example, "cluster.local/ns/default/sa/bookinfo-ratings-v2" will return "cluster.local"
+func extractTrustDomainFromAuthzPrincipal(principal string) string {
```

Comment:
just call this `getTrustDomain` and ideally this should be added in pkg/spiffe/spiffe.go.

Reply:
I'll change the name, but the authz principals do not have true spiffe format so let's keep it in here for now?

Reply:
yeah, I'm fine to keep it here for now.